.github/workflows/ci.yml

```diff
@@ -0,0 +1,108 @@
+# This workflow will build a Java project with Maven
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-maven
+
+name: XNIO CI
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+
+jobs:
+  build-all:
+    name: Compile (no tests) with JDK 8
+    runs-on: ubuntu-latest
+    steps:
+    - uses: n1hility/cancel-previous-runs@v2
+      with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+    - uses: actions/cache@v1
+      with:
+        path: ~/.m2/repository
+        key: m2-${{ hashFiles('**/pom.xml') }}
+        restore-keys: |
+            m2-
+    - uses: actions/checkout@v2
+    - name: Host information
+      run: netstat -anp
+      shell: bash
+    - name: Set up JDK 8
+      uses: joschi/setup-jdk@v2
+      with:
+        java-version: 8
+    - name: Build
+      run: mvn -U -B -fae -DskipTests -Dfindbugs clean install
+    - name: Tar Maven Repo
+      shell: bash
+      run: tar -czf maven-repo.tgz -C ~ .m2/repository
+    - name: Persist Maven Repo
+      uses: actions/upload-artifact@v1
+      with:
+        name: maven-repo
+        path: maven-repo.tgz
+    - uses: actions/upload-artifact@v2
+      if: failure()
+      with:
+        name: surefire-reports-build
+        path: '**/surefire-reports/*.txt'
```

Comment:
AFAIK this stuff isn't needed, is it?

Reply:
Well the approach here is one runner is building then its tarring the resulting repo and all of the other runners are just running mvn test. Overall its quite a bit faster than actually building & testing on each runner, but up to you which way you'd prefer.

Reply:
OK I suppose it makes sense given the massive testing matrix :grin:...